### ISSUE TITLE
color-themes not working

### ISSUE THREAD

**Issue by AdrienLemaire** [felixr/urxvt-theme-switch#1]
I tried to set up urxvt-theme-switch, but nothing happen and I can't figure out why (no good with perl).

Here is the related part in my .Xresources:

```
URxvt*termName: rxvt-unicode-256color
URxvt.perl-lib: /home/dori/Projects/3rd-part/urxvt-perls/
URxvt.perl-ext:             default
URxvt.perl-ext-common:      default,vtwheel,url-select,keyboard-select,resize-font,color-themes
URxvt.color-themes.dir: ~/.colors2/
URxvt.intensityStyles: false

URxvt.keysym.C-M-n:           perl:color-themes:next
URxvt.keysym.C-M-p:           perl:color-themes:prev
```

Note that I made a symlink to color-themes from the `urxvt-perls/` directory so urxvt can access the script from **URxvt.perl-lib**.
Other plugins like **resize-font** work fine. **color-themes** doesn't and no error is reported.

Here is one of the themes I have in my `~/.colors2/` directory: http://sprunge.us/BLUF (I only need 2 themes, light and dark solarized).

Can you figure out why it's not working? Thanks in advance

**Comment by felixr** [comment]
I think there are two issues here:
- Using ~ (tilde) to refer to the home directory does not work, even though the source code suggests that this should work. Try using the absolute path to your theme dir
- the themes are not pre-processed the way .Xdefaults are. So you have to run the pre-processor yourself if you want to use #define (e.g: cpp BLUF > bluff)

**Comment by felixr** [comment]
I fixed the problem with ~ in paths in 610cb619c851445f1faede5c789f9d5abdf14626 and I am also showing a message if no color schemes were loaded (85d86b75aecb3c13bc9f5405266daf000a68eb05). Please try the updated version and tell me if it works for you.

**Comment by AdrienLemaire** [comment]
Thank you for your fast answer :)

The message is quite useful, I now know that I'm loading the correct file with the correct map key.
I also hardcoded colors instead of using #define, and the switch theme is working!

I'm now wondering if there is a way to call active urxvt windows from a script and call that color-themes script to update all urxvt windows colors on the fly.  It would go in [that larger script](http://sprunge.us/AeTO?sh).

Do you think it's possible?

Closing the issue, thank you very much
